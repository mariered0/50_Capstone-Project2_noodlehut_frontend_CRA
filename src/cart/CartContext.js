import React from "react";

const cartContext = React.createContext();

export default cartContext;
cartContext.displayName = "Cart";