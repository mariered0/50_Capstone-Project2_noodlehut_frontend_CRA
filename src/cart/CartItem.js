import React, { useState, useContext, useEffect } from "react";
import cartContext from "./CartContext";
import { Box, Grid, Avatar, Typography, Divider, IconButton, Stack } from "@mui/material";
import CloseIcon from "@mui/icons-material/Close";
import AddCircleOutlineIcon from "@mui/icons-material/AddCircleOutline";
import RemoveCircleOutlineIcon from "@mui/icons-material/RemoveCircleOutline";
import img from "../static/image/image_unavailable.png";

const CartItem = ({ id, name, qty, price }) => {
  const { deleteItem, addToCart, removeFromCart } =
    useContext(cartContext);

  return (
    <Box key={id} sx={{ flexGrow: 1 }}>
      <Grid container>
        <Grid item xs={3} >
          <Avatar
            src={img}
            sx={{ mr: 2,
                  width: { xs: 0,
                           sm: 0,
                           md: 80,
                           lg: 80,
                           xl: 80},
                  height: { xs: 0,
                            sm: 0,
                            md: 80,
                            lg: 80,
                            xl: 80}
                  }}
          />
        </Grid>
        <Grid item xs={7}>
          <Typography variant="h6" align="left"> {name}</Typography>
          <Box display="flex" justifyContent={"space-between"}>
            <Typography variant="body1">
              {qty} total
            </Typography>
            <Typography variant="body1">
              ${price}
            </Typography>
          </Box>
        </Grid>

        <Grid item xs={1}>
          <Stack>
            <IconButton aria-label="add one" onClick={() => addToCart({id, name, price})}>
              <AddCircleOutlineIcon />
            </IconButton>
            <IconButton
              aria-label="delete one"
              onClick={() => removeFromCart(id)}
            >
              <RemoveCircleOutlineIcon />
            </IconButton>
          </Stack>
        </Grid>
        <Grid item xs={1} >
          <Box sx={{ height: 80 }} display="flex" alignItems="center"> 
          <IconButton aria-label="delete" onClick={() => deleteItem(id)}>
            <CloseIcon />
          </IconButton>
          </Box>

        </Grid>
      </Grid>
      <Divider variant="inset" />
    </Box>
  );
};

export default CartItem;
