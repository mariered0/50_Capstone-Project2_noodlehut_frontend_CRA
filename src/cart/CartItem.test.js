import React from "react";
import { render } from '@testing-library/react';
import CartContext from "./CartContext";
import CartItem from "./CartItem";
const { deleteItem, addToCart, removeFromCart } = require("./_testCommon");

it("renders without crashing", () => {
    render(
        <CartContext.Provider value={{ deleteItem, addToCart, removeFromCart }}>
            <CartItem />
        </CartContext.Provider>
    );
});

it("matches snapshot", () => {
	const { asFragment } = render(
        <CartContext.Provider value={{ deleteItem, addToCart, removeFromCart }}>
            <CartItem />
        </CartContext.Provider>
    
    );
	expect(asFragment()).toMatchSnapshot();
});

