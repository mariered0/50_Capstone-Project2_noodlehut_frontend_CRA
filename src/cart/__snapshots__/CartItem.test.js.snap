// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-gy4mli-MuiAvatar-root"
        >
          <img
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="image_unavailable.png"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft css-1lv9nvb-MuiTypography-root"
        >
           
        </h6>
        <div
          class="MuiBox-root css-gg4vpm"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
             total
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            $
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
      >
        <div
          class="MuiStack-root css-nen11g-MuiStack-root"
        >
          <button
            aria-label="add one"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AddCircleOutlineIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="delete one"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="RemoveCircleOutlineIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-15oeajw"
        >
          <button
            aria-label="delete"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-inset css-uls1yk-MuiDivider-root"
    />
  </div>
</DocumentFragment>
`;
